netshare/utils/field.py

```diff
@@ -191,6 +214,57 @@ def getOutputDim(self) -> int:
         return 2 * self.num_bits  # type: ignore
 
 
+class Word2VecField(Field):
+    def __init__(self, word2vec_size, word2vec_cols, *args, **kwargs):
+        super(Word2VecField, self).__init__(*args, **kwargs)
+
+        self.word2vec_size = word2vec_size
+        self.norm_option = Normalization.MINUSONE_ONE
+        self.dict_encoding_type_vs_cols = get_word2vec_type_col(word2vec_cols)
+
+    def normalize(self, x, embed_model):
+        return np.array(
+            [get_vector(embed_model, str(xi), norm_option=True) for xi in x]
+        )
+
+    def denormalize(self, norm_x):
+        dict_annDictPair = {}
+        with open(get_annoy_dict_idx_ele_for_word2vec(), "r") as readfile:
+            dict_annDictPair = json.load(readfile)
+        dict_annoyIndex = {}
+        for encoding_type in dict_annDictPair:
+            type_ann = AnnoyIndex(self.word2vec_size, "angular")
+            type_ann.load(get_annoyIndex_for_word2vec(encoding_type))
+            dict_annoyIndex[encoding_type] = type_ann
+
+        for k in self.dict_encoding_type_vs_cols:
+            if self.name in self.dict_encoding_type_vs_cols[k]:
+                encoding_type = k
+                break
+        else:
+            raise ValueError("Cannot find the word2vec key!")
+        # When constructing the dict_annDictPair, dict_annDictPair[encoding_type]: dict{}
+        # is a k v pair where k is an integer. After using Json.save() then Json.load(), it
+        # will change k into str. Therefore, when we are using dict_annDictPair[encoding_type]
+        # the k should be casted to int.
```

Comment:
sounds like an awful bug 😂 
Well done!!!
Here's a relevant link: https://stackoverflow.com/questions/1450957/pythons-json-module-converts-int-dictionary-keys-to-strings